<a href="https://colab.research.google.com/github/Neoneto/CodingDojo_Week3/blob/main/Average_Height.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Average Height Exercise 
Submitted by Kenneth Alaba

## Load the data

In [42]:
# Import the pandas library
import pandas as pd


In [43]:
# Load the data

## Mounting Google Drive
from google.colab import drive
drive.mount('/content/drive')

## Opening the file
filename = '/content/drive/My Drive/Coding Dojo/03 Week 3: Exploratory Viz/athleteEventsNoPersonal.csv'

## Storing the data in olympic_df
olympic_df = pd.read_csv(filename)

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


## Inspect the dataframe

In [44]:
# Show the first few rows
olympic_df.head()

ID   Age  Height  ...          Sport                               Event Medal
0   5  21.0   185.0  ...  Speed Skating    Speed Skating Women's 500 metres   NaN
1   5  21.0   185.0  ...  Speed Skating  Speed Skating Women's 1,000 metres   NaN
2   5  25.0   185.0  ...  Speed Skating    Speed Skating Women's 500 metres   NaN
3   5  25.0   185.0  ...  Speed Skating  Speed Skating Women's 1,000 metres   NaN
4   5  27.0   185.0  ...  Speed Skating    Speed Skating Women's 500 metres   NaN

[5 rows x 12 columns]

In [45]:
# Show the info of the dataframe
olympic_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40616 entries, 0 to 40615
Data columns (total 12 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   ID      40616 non-null  int64  
 1   Age     39143 non-null  float64
 2   Height  31615 non-null  float64
 3   Team    40616 non-null  object 
 4   NOC     40616 non-null  object 
 5   Games   40616 non-null  object 
 6   Year    40616 non-null  int64  
 7   Season  40616 non-null  object 
 8   City    40616 non-null  object 
 9   Sport   40616 non-null  object 
 10  Event   40616 non-null  object 
 11  Medal   5917 non-null   object 
dtypes: float64(2), int64(2), object(8)
memory usage: 3.7+ MB


We can see that the dataframe has 12 columns and 40616 rows. From the info, we can also see that the columns 'Age' and 'Medal' has missing values. Additionally, the column 'Medal' has a significant number of missing values. This is because the dataset contains info of all the olympic participants, regardless whether they won or not, but not all has won a medal. Those without medals result in a missing value

## Participants from London

In [46]:
# Creates the filter for the city
london_filter = olympic_df['City'] == 'London'

# Apply the filter and store the filtered data in a separate dataframe
london_df = olympic_df.loc[london_filter]

# Count the number of participants from london
len(london_df['City'])

3370

## Minimum age

In [47]:
# Get the minimum value in the 'Age' column
london_df['Age'].min()

15.0

## Tug-of-War participants from Sweden

In [48]:
# Creates the filter for the sport event and the team
sport_filter = olympic_df['Sport'] ==  'Tug-Of-War'
team_filter = olympic_df['Team'] == 'Sweden'

# Apply the filters and count the resulting data
len(olympic_df.loc[sport_filter & team_filter])

6

## Gold medals by 40+ yr old participants

In [49]:
# Creates filter for 'Age' and 'Medal'
age_filter = olympic_df['Age'] > 40 #(above 40,exclusive)
medal_filter = olympic_df['Medal'] == 'Gold'

# Counts the number of the resulting data
len(olympic_df[age_filter & medal_filter])

66

## Average Height

### Including all rows

In [50]:
# Gets the mean of the 'Height' column
# Rounds the result to 2 decimal places 
round(olympic_df['Height'].mean(), 2)

175.38

### Grouped by 'Event'

In [51]:
# Group the data by the 'Event' and 
olympic_df.groupby('Event')['Height'].mean()

Event
Alpine Skiing Men's Combined                      178.547170
Alpine Skiing Men's Downhill                      178.211679
Alpine Skiing Men's Giant Slalom                  176.739394
Alpine Skiing Men's Slalom                        176.482143
Alpine Skiing Men's Super G                       179.850575
                                                     ...    
Wrestling Women's Flyweight, Freestyle            158.100000
Wrestling Women's Heavyweight, Freestyle          173.444444
Wrestling Women's Light-Heavyweight, Freestyle    170.000000
Wrestling Women's Lightweight, Freestyle          162.666667
Wrestling Women's Middleweight, Freestyle         164.533333
Name: Height, Length: 726, dtype: float64

### Dropping duplicate athletes

In [52]:
# Creates a new dataframe where duplicate athletes are dopped
one_per_athlete = olympic_df.drop_duplicates(subset = 'ID', keep = 'first')

# calculate the mean of the 'Height' column
ave_unique_height = one_per_athlete['Height'].mean()

# Show the result, rounded to 2 decimal places
round(ave_unique_height, 2)

176.32